File: src/pattern/composite/java/Entry.java
package pattern.composite.java;

public abstract class Entry {
    abstract String getName();
    abstract int getSize();
    public void printList(){
        printList("");
    }
    protected abstract void printList(String prefix);

    public Entry add(Entry entry) throws FileTreatmentException {
        throw new FileTreatmentException();
    }

    @Override
    public String toString() {
        return this.getName()+"("+this.getSize()+")";
    }
}

File: src/pattern/composite/kotlin/Entry.kt
package pattern.composite.kotlin

abstract class Entry {
    abstract fun getName(): String
    abstract fun getSize(): Int
    abstract fun printList(name: String)
    fun printList(){
        this.printList("")
    }

    open fun add(entry: Entry): Entry{
        throw FileTreatmentException()
    }

    override fun toString(): String {
        return "${this.getName()}(${this.getSize()})"
    }



}